### ISSUE TITLE
Question about Training Time

### ISSUE THREAD

**Issue by accountspark-bo-peng** [mit-han-lab/data-efficient-gans#15]
Hi,

Great good. Congrats. 

I would like to train your model on this fruits dataset: [https://www.kaggle.com/moltean/fruits](https://www.kaggle.com/moltean/fruits )

It has about 80K images, 100 classes. 

How long do you think it would take to train on Colab GANs for all classes? Does the model train one class at a time or does it train multiple classes all at once?

Thanks for the help and great work!

**Comment by zsyzzsoft** [comment]
The StyleGAN2 model is mainly designed for unconditional generation, so either training one class at a time or training all classes regardless of the labels is supported currently. You may also try the BigGAN model, which is for conditional generation. As the dataset is quite large, it may take long to converge. I'm afraid that training on Colab may take days or even weeks.

**Comment by accountspark-bo-peng** [comment]
Ok really good job.

So your repo is just image augmentation right? If I want to try to implement a similar thing on a basic generator/discriminator model where should I put the augmentation?

Should I do both augmentation on the training image and the generated image?

Thanks.

**Comment by zsyzzsoft** [comment]
Yes, augment both reals and fakes, during both generator and discriminator training. We also explain this (here)[https://github.com/mit-han-lab/data-efficient-gans#using-diffaugment-for-your-own-training].

**Comment by accountspark-bo-peng** [comment]
Sounds good. You probably can make it even better right if you try other augmentation like shift scale rotate, cutmix, mixin etc.. right?

**Comment by zsyzzsoft** [comment]
Not every augmentation works. We have at least tried scaling and rotation and didn't found them helpful. I recommend starting with our given augmentations. Besides, even though the entire training may take long, I still recommend beginning with our DiffAugment-stylegan2 or DiffAugment-biggan repo - you will see reasonable results in a likely short time.

**Comment by BoPengGit** [comment]
Hi,

Great job. Yea like I mentioned, I think you can improve the results trying other things such as Mixin, cutmix etc. 

Hey I'm trying to create a GIF like yours. Can you tell me in plain english how to do this? Do I generate a bunch of images of the same class from a bunch of random noise vectors, and then I have to sort the generated images somehow for a smooth transition? 

Thanks and good work.

**Comment by zsyzzsoft** [comment]
Yes, we have provided [generate_gif.py](https://github.com/mit-han-lab/data-efficient-gans/blob/master/DiffAugment-stylegan2/generate_gif.py) as explained [here](https://github.com/mit-han-lab/data-efficient-gans/tree/master/DiffAugment-stylegan2#generating-an-interpolation-video).

**Comment by BoPengGit** [comment]
Thanks, I think we can close if you want, Good work.
